src/ttsim/convert_nested_data.py

```diff
@@ -87,8 +119,51 @@ def create_data_tree_from_df(
     return dt.unflatten_from_qual_names(name_to_input_series)
 
 
+def _fail_if_noncompatible_objects_in_nested_data(
+    paths_to_data: QualNameData,
+) -> None:
+    """Fail if the nested data contains non-compatible objects."""
+    faulty_paths = []
+    for path, data in paths_to_data.items():
+        if isinstance(data, (pd.Series, np.ndarray, list)):
+            if all(isinstance(item, (int, float, bool)) for item in data):
+                continue
+            else:
+                faulty_paths.append(str(path))
+        elif isinstance(data, (int, float, bool)):
+            continue
+        else:
+            faulty_paths.append(str(path))
+    if faulty_paths:
+        msg = format_errors_and_warnings(
+            "The data returned contains objects that cannot be casted to "
+            "a pandas.DataFrame. Make sure that the requested targets return scalars "
+            "(int, bool, float) only."
+            "The following paths contain non-scalar objects: "
+            f"{format_list_linewise(faulty_paths)}"
+        )
+        raise TypeError(msg)
+
+
+def _fail_if_data_paths_are_missing_in_paths_to_column_names(
+    available_paths: list[str],
+    required_paths: list[str],
+) -> None:
+    """Fail if the data paths are missing in the paths to column names."""
+    missing_paths = [
+        str(path) for path in required_paths if path not in available_paths
+    ]
+    if missing_paths:
+        msg = format_errors_and_warnings(
+            "Converting the nested data to a DataFrame failed because the following "
+            "paths are not mapped to a column name: "
+            f"{format_list_linewise(list(missing_paths))}"
+        )
+        raise ValueError(msg)
+
+
 def _fail_if_mapper_has_incorrect_format(
```

Comment:
Apologies I notice only now, but I think the first part of this just duplicates `dt.fail_if_paths_are_invalid`? Can we re-use that instead?